packages/fxa-settings/src/components/Settings/UnitRowTwoStepAuth/index.tsx

```diff
@@ -21,213 +21,239 @@ export const UnitRowTwoStepAuth = () => {
   const alertBar = useAlertBar();
   const account = useAccount();
   const {
-    // TODO: use in FXA-10206
-    // backupCodes: { hasBackupCodes, count },
+    backupCodes: { count },
     totp: { exists, verified },
   } = account;
-  const [modalRevealed, revealModal, hideModal] = useBooleanState();
-  const [secondaryModalRevealed, revealSecondaryModal, hideSecondaryModal] =
+  const [disable2FAModalRevealed, revealDisable2FAModal, hideDisable2FAModal] =
     useBooleanState();
-  const { l10n } = useLocalization();
+  const [
+    newBackupCodesModalRevealed,
+    revealNewBackupCodesModal,
+    hideNewBackupCodesModal,
+  ] = useBooleanState();
+  const ftlMsgResolver = useFtlMsgResolver();
 
   const disableTwoStepAuth = useCallback(async () => {
     try {
       await account.disableTwoStepAuth();
-      hideModal();
+      hideDisable2FAModal();
       alertBar.success(
-        l10n.getString(
+        ftlMsgResolver.getMsg(
           'tfa-row-disabled-2',
-          null,
           'Two-step authentication disabled'
         )
       );
     } catch (e) {
-      hideModal();
+      hideDisable2FAModal();
       alertBar.error(
-        l10n.getString(
+        ftlMsgResolver.getMsg(
           'tfa-row-cannot-disable-2',
-          null,
           'Two-step authentication could not be disabled'
         )
       );
     }
-  }, [account, hideModal, alertBar, l10n]);
+  }, [account, hideDisable2FAModal, alertBar, ftlMsgResolver]);
 
-  const conditionalUnitRowProps =
+  const DisableTwoStepAuthModal = () => {
+    return (
+      <VerifiedSessionGuard
+        onDismiss={hideDisable2FAModal}
+        onError={(error) => {
+          hideDisable2FAModal();
+          alertBar.error(error.message, error);
+        }}
+      >
+        <Modal
+          onDismiss={hideDisable2FAModal}
+          onConfirm={() => disableTwoStepAuth()}
+          headerId="two-step-auth-disable-header"
+          descId="two-step-auth-disable-description"
+          confirmText={ftlMsgResolver.getMsg(
+            'tfa-row-disable-modal-confirm',
+            'Disable'
+          )}
+          confirmBtnClassName="cta-caution cta-base-p"
+        >
+          <FtlMsg id="tfa-row-disable-modal-heading">
+            <h2
+              className="font-bold text-xl text-center mb-2"
+              data-testid="disable-totp-modal-header"
+            >
+              Disable two-step authentication?
+            </h2>
+          </FtlMsg>
+          {/* "replacing backup authentication codes" link below will actually drop you into
+          backup authentication codes flow in the future. */}
+          <FtlMsg
+            id="tfa-row-disable-modal-explain-1"
+            elems={{
+              linkExternal: (
+                <LinkExternal
```

Comment:
Just another nit, you could set `LinkExternal` here to a variable and then you can reference it here and in the fallback, just to keep them consistent. (we do this in `CardHeader` with `spanElement`)